src/main/java/org/apache/bcel/generic/MethodGen.java

```diff
@@ -599,27 +599,33 @@ public void removeCodeAttributes() {
     /**
      * @since 6.0
      */
-    public void addAnnotationsAsAttribute(final ConstantPoolGen cp) {
-          final Attribute[] attrs = AnnotationEntryGen.getAnnotationAttributes(cp, super.getAnnotationEntries());
+    public Attribute[] addAnnotationsAsAttribute(final ConstantPoolGen cp) {
```

Comment:
-1: You cannot change a return type and maintain binary compatibility.
Run `mvn clean install -DskipTests japicmp:cmp` and see the compatibility break:
```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  23.786 s
[INFO] Finished at: 2019-10-11T15:14:52-04:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.github.siom79.japicmp:japicmp-maven-plugin:0.14.1:cmp (default-cli) on project bcel: There is at least one incompatibility: org.apache.bcel.generic.MethodGen.addAnnotationsAsAttribute(org.apache.bcel.generic.ConstantPoolGen):METHOD_RETURN_TYPE_CHANGED,org.apache.bcel.generic.MethodGen.addParameterAnnotationsAsAttribute(org.apache.bcel.generic.ConstantPoolGen):METHOD_RETURN_TYPE_CHANGED -> [Help 1]
[ERROR]
```
You need a _new_ API.